src/EditorFeatures/Core/Shared/Tagging/EventSources/TaggerEventSources.TextChangedEventSource.cs

```diff
@@ -10,34 +11,34 @@ internal partial class TaggerEventSources
     {
         private class TextChangedEventSource : AbstractTaggerEventSource
         {
-            private readonly ITextBuffer _subjectBuffer;
+            private readonly ITextBuffer2 _subjectBuffer;
 
             public TextChangedEventSource(ITextBuffer subjectBuffer, TaggerDelay delay)
                 : base(delay)
             {
                 Contract.ThrowIfNull(subjectBuffer);
 
-                _subjectBuffer = subjectBuffer;
+                _subjectBuffer = (ITextBuffer2)subjectBuffer;
             }
 
             public override void Connect()
             {
-                _subjectBuffer.Changed += OnTextBufferChanged;
+                _subjectBuffer.ChangedAsync += OnTextBufferChanged;
             }
 
             public override void Disconnect()
             {
-                _subjectBuffer.Changed -= OnTextBufferChanged;
+                _subjectBuffer.ChangedAsync -= OnTextBufferChanged;
```

Comment:
Are there still possibility of races where we get OnTextBufferChanged after this unsubscribe?